server/WebAPI/Controllers/ScheduledPlaceToEatController.cs

```diff
@@ -0,0 +1,118 @@
+    }
+
+    // GET: api/ScheduledPlaceToEat
+    [HttpGet]
+    [ProducesResponseType(typeof(IEnumerable<InputScheduledPlaceToEatForCreateDTO>), StatusCodes.Status200OK)]
+    public IActionResult Get()
+    {
+      return Ok(_scheduledPlaceToEatService.GetEatingByUserId(HttpContext.GetUserIdFromClaim()));
+    }
+
+    [HttpPost("UploadEatingMultiFile/{id}")]
+    [Consumes("multipart/form-data")]
+    public IActionResult UploadEatingMultiFile([FromForm] IFormFileCollection files, int id)
+    {
+      if(files.Count <= Consts.MaxEatingFileCount && files.Count > 0)
+      {
+        foreach(var file in files)
+        {
+          if(!Consts.AllowedImageContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase) && !Consts.AllowedTextContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase))
+          {
+            return BadRequest("notAllowedContentType");
+          }
+
+          if(file.Length > Consts.MaxEatingFileSize)
+          {
+            return BadRequest("fileIsToBig");
+          }
+
+          _attachmentFileEatingService.Create(file, id, HttpContext.GetUserIdFromClaim());
+        }
+      }
+      return Ok();
+    }
+
+    [HttpPost("DeleteEatingFile")]
+    public IActionResult DeleteEatingFiles(int fileId)
+    {
+      _attachmentFileEatingService.Delete(fileId, HttpContext.GetUserIdFromClaim());
+      return Ok();
+    }
+
+    // POST: api/ScheduledPlaceToEat/Create
+    [HttpPost("Create")]
+    public int Create([FromBody] InputScheduledPlaceToEatForCreateDTO ScheduledPlaceToEatDTO)
+    {
+      ScheduledPlaceToEatDTO.UserId = HttpContext.GetUserIdFromClaim();
+      return _scheduledPlaceToEatService.CreateNewEating(ScheduledPlaceToEatDTO);
+    }
+
+    // POST: api/ScheduledPlaceToEat/Update
+    [HttpPost("Update")]
+    public IActionResult Update([FromBody] InputScheduledPlaceToEatForUpdateOrDeleteDTO ScheduledPlaceToEatDTO)
+    {
+      ScheduledPlaceToEatDTO.UserId = HttpContext.GetUserIdFromClaim();
+      return _scheduledPlaceToEatService.UpdateScheduledPlaceToEat(ScheduledPlaceToEatDTO) ? Ok() : (IActionResult)BadRequest();
+    }
+
+    // DELETE: api/ScheduledPlaceToEat/Delete
+    [HttpPost("Delete")]
+    public IActionResult Delete([FromBody] InputScheduledPlaceToEatForUpdateOrDeleteDTO ScheduledPlaceToEatDTO)
+    {
+      ScheduledPlaceToEatDTO.UserId = HttpContext.GetUserIdFromClaim();
+      return _scheduledPlaceToEatService.DeleteScheduledPlaceToEat(ScheduledPlaceToEatDTO) ? Ok() : (IActionResult)BadRequest();
+    }
+
+    // DELETE: api/ScheduledPlaceToEat/DeleteById/id
+    [HttpPost("DeleteById/{id}")]
+    public IActionResult Delete( int id)
+    {
+      var scheduledPlaceToEatDTO = _scheduledPlaceToEatService.GetEatingByUserId(HttpContext.GetUserIdFromClaim()).Where(r => r.Id.Equals(id)).FirstOrDefault();
+      return Delete(new InputScheduledPlaceToEatForUpdateOrDeleteDTO {
```

Comment:
i really don't understand why you need for deleting and updating dto

Reply:
the id property is optional for creating a dto. for dto, the id property is required for updating or deleting. but since I now only use the id from Route for deletion, this model will not be used for deletion, only for update